src/main/java/com/enonic/lib/react4xp/ssr/ServerSideRenderer.java

```diff
@@ -30,291 +21,136 @@
-                    "in " + ServerSideRenderer.class.getName() + ".loadAsset\n" +
-                    "assetName = '" + assetName + "'\n" +
-                    "resource url = '" + url + "'\n" +
-                    getSolutionTips());
-
-            if (IS_PRODMODE) {
-                ASSET_CACHE_MARKERS.get(assetName).isCached = false;
+                // Why sleep? When initializing multiple Renderers at once, this prevents them from borrowing one and returning
+                // that before the next one is borrowed: if a non-destroyed Renderer is cycled like that, nothing is actually initialized.
+                Thread.sleep(3);
 
+            } catch (Exception e) {
+                e.printStackTrace();
             }
-            throw new RenderException(e1.getMessage());
-
-        } catch (ScriptException e2) {
-            String cleanErrorMessage = getCleanErrorMessage(e2);
-            LOG.error(
-                    getLoggableStackTrace(e2, cleanErrorMessage) + "\n\n" +
-                    e2.getClass().getSimpleName() + ": " + cleanErrorMessage + "\n" +
-                    "in " + ServerSideRenderer.class.getName() + ".loadAsset\n" +
-                    "assetName = '" +assetName + "'\n" +
-                    "resource url = '" + url + "'\n" +
-                    getSolutionTips());
-
-            if (IS_PRODMODE) {
-                ASSET_CACHE_MARKERS.get(assetName).isCached = false;
-            }
-            throw new RenderException(cleanErrorMessage);
-        }
-    }
-
-
-    private Map<String, String> runSSR(String entry, String props, LinkedList<String> assetsInvolved) throws ScriptException {
 
-        String callScript = "var obj = { " +
-                KEY_HTML + ": ReactDOMServer.renderToString(" + LIBRARY_NAME  + "['" + entry + "'].default(" + props  + ")) " +
-                "};" +
-                "obj;";
-
-
-        try {
-            ScriptObjectMirror obj = (ScriptObjectMirror)ENGINE.eval(callScript);
-
-            String rendered = (String)obj.get(KEY_HTML);
-
-            return Map.of( KEY_HTML, rendered );
-
-        } catch (ScriptException e) {
-            String cleanErrorMessage = getCleanErrorMessage(e);
-            LOG.error(
-                    getLoggableStackTrace(e, cleanErrorMessage) + "\n\n" +
-                    e.getClass().getSimpleName() + ": " + cleanErrorMessage + "\n" +
-                    "in " + ServerSideRenderer.class.getName() + ".runSSR\n" +
-                    "entry = '" + entry + "'\n" +
-                    "Failing call: '" + callScript + "'\n" +
-                    getSolutionTips());
-
-            if (IS_PRODMODE && assetsInvolved != null) {
-                for (String asset : assetsInvolved) {
-                    CacheMarker cacheMarker = ASSET_CACHE_MARKERS.get(asset);
-                    synchronized (cacheMarker) {
-                        cacheMarker.isCached = false;
-                    }
-                }
+            if (renderer != null) {
```

Comment:
should be in `finally` block

Reply:
Done - this and the other one.

Out of curiosity: this is after a `catch`, which should catch _all exceptions_ and just log them before proceeding - should be no `return` or other flow changes inside the catch block, so you'd expect the `if (renderer!=null) {` etc will run no matter what, with or without a finally block - _unless an error occurs during the catch block itself._ 

Is this - errors my happen during the catch but the finally will run no matter what - the only reason for using a finally block here, or are there other best-practice reasons too, even in this case?